### PR TITLE
Move `numeric_literals.md` design doc into `expressions/literals.md` and update

### PR DESCRIPTION
**Pull request by josh11b**
Follow-on to change #2410 which created the `expressions/literals.md` home for literal expressions. Incorporates the decision in #2113 to address a TODO.

### CODE REVIEW

File: docs/design/README.md [hunk 1]
```diff
@@ -443,11 +443,11 @@ programming error:
     to a mathematically incorrect result, such as a two's complement result or
     zero.
 
-The unsigned-integer types are written `uN` or `Carbon.UInt(N)`, with `N` a
-positive multiple of 8. Unsigned integer types wrap around on overflow; we
-strongly advise that they are not used except when those semantics are desired.
-These types are intended for bit manipulation or modular arithmetic as often
-found in [hashing](https://en.wikipedia.org/wiki/Hash_function),
+The unsigned-integer types are written `uN`, with `N` a positive multiple of 8.
```
Comment (zygoloid) [inline docs/design/README.md]: The contrast between "may be written" above (with an example that `i32` is the same as some library type) and "are written" here suggests to me that `iN` is an alias for a library type but `uN` is not. Can we make the phrasing here consistent?

File: docs/design/README.md [hunk 2]
```diff
@@ -427,9 +427,9 @@ and [`while`](#while), and
 
 ### Integer types
 
-The signed-integer type with bit width `N` may be written `iN` or
-`Carbon.Int(N)`, as long as `N` is a positive multiple of 8. For example, `i32`
-is equivalent to `Carbon.Int(32)`. Signed-integer
+The signed-integer type with bit width `N` may be written `iN`, as long as `N`
+is a positive multiple of 8. For example, `i32` is equivalent to
+`Carbon.Int(32)`. Signed-integer
```
Comment (zygoloid) [inline docs/design/README.md]: Should this say `Core.Int(32)`?

File: docs/design/README.md (hunk 1 above)
Comment (josh11b) [reply docs/design/README.md]: I made them both say "may be written"

File: docs/design/README.md (hunk 2 above)
Comment (josh11b) [reply docs/design/README.md]: I rewrote to not mention the `Core.Int` type. That is a detail that I think is fine being only in `expressions/literals.md`.

**Review by zygoloid — APPROVED** [review]

**Pull request merged**